recipes-extended/xen/files/libxl-stubdom-nvram.patch

```diff
@@ -0,0 +1,27 @@
+Index: xen-4.6.4/tools/libxl/libxl_dm.c
+===================================================================
+--- xen-4.6.4.orig/tools/libxl/libxl_dm.c
++++ xen-4.6.4/tools/libxl/libxl_dm.c
+@@ -1159,11 +1159,17 @@ static int libxl__build_device_model_arg
+                  * For other disks we translate devices 0..3 into
+                  * hd[a-d] and ignore the rest.
+                  */
+-                if (strncmp(disks[i].vdev, "sd", 2) == 0)
+-                    drive = libxl__sprintf
+-                        (gc, "file=%s,if=scsi,bus=0,unit=%d,format=%s,cache=writeback",
+-                         pdev_path, disk, format);
+-                else if (disk < 6 && b_info->u.hvm.hdtype == LIBXL_HDTYPE_AHCI) {
++                if (strncmp(disks[i].vdev, "sd", 2) == 0) {
++                    if (b_info->stubdomain_version == LIBXL_STUBDOMAIN_VERSION_LINUX) {
++                        drive = libxl__sprintf
++                            (gc, "file=%s%c,if=scsi,bus=0,unit=%d,format=%s,cache=writeback",
++                             "/dev/xvd", 'a'+disk, disk, format);
```

Comment:
Is there any way not to have to hard code the device path? Maybe make it a #define variable?